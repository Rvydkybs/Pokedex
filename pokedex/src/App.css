.container {
  flex: 1;
}
.innerContainer {
  background-color: pink;
}
.textContainer {
}

.imageContainer {
}
