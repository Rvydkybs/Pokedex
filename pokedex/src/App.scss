h1 {
  font-weight: 600;
  font-size: 55px;
  text-align: center;
  color: white;
}
