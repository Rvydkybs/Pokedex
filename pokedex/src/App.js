import "./App.scss";
import React, { useEffect, useState, useMemo } from "react";
// import { Button } from "@material-ui/core";
import { Link } from "react-router-dom";
import { Button } from "@mui/material";

import Pagination from "./Components/Pagination";

let PageSize = 10;
var index,
  indexList = [];

function App() {
  const [pokemons, setPokemon] = useState({
    pokeItems: [],
    isDataLoaded: false,
  });
  const [currentPage, setCurrentPage] = useState(1);
  const [pokemonIndex, setPokemonIndex] = useState();

  useEffect(() => {
    fetch("https://pokeapi.co/api/v2/pokemon?limit=150")
      .then((res) => res.json())
      .then((json) => {
        setPokemon({ pokeItems: json, isDataLoaded: true });
        console.log("deneme2", pokemons.pokeItems);
      });
    console.log("deneme", pokemons.pokeItems);
  }, []);

  useEffect(() => {
    fetch("https://pokeapi.co/api/v2/pokemon?limit=150")
      .then((res) => res.json())
      .then((json) => {
        setPokemon({ pokeItems: json, isDataLoaded: true });
        console.log("deneme2", pokemons.pokeItems);
      });
    console.log("deneme", pokemons.pokeItems);
  }, [currentPage]);

  const currentTableData = useMemo(() => {
    const firstPageIndex = (currentPage - 1) * PageSize;
    const lastPageIndex = firstPageIndex + PageSize;

    index =
      pokemons.pokeItems.results &&
      pokemons.pokeItems.results.slice(firstPageIndex, lastPageIndex);
    index &&
      index.map((item, index) => {
        //if i had to use original url for images;
        // var url = "https://pokeapi.co/api/v2/pokemon/";
        // url.substring(0, url.length - 1); //delete '/' from url to add .png
        // const imageUrl = url + ".png";
        // imageUrl.split(" ");

        indexList.push(
          `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${item.url.slice(
            34,
            36
          )}.png`
        );
        console.log("dddd", indexList);
      });

    return (
      pokemons.pokeItems.results &&
      pokemons.pokeItems.results.slice(firstPageIndex, lastPageIndex)
    );
  }, [currentPage]);

  return (
    <div
      style={{
        backgroundColor: "#0C0A3E",
        flex: 1,
        display: "flex",
        flexDirection: "column",
      }}
    >
      <h1>
        <b>Pokedex</b>
      </h1>
      <div
        style={{
          width: "100%",
          flexDirection: "column",
          display: "flex",
          alignItems: "center",
        }}
      >
        {currentTableData &&
          indexList &&
          currentTableData.map((item, index) => {
            const linkContent = indexList[index];
            return (
              <div
                style={{
                  margin: 15,
                  padding: 15,
                  width: 400,
                  display: "flex",
                  justifyContent: "center",
                  borderRadius: 12,
                  borderWidth: 2,
                  backgroundColor: "#3c6186",
                  borderColor: "#3c6186",
                  marginTop: 30,
                }}
              >
                <img
                  width={200}
                  height={200}
                  className="imageContainer"
                  src={linkContent}
                />
                <div
                  style={{
                    display: "flex",
                    justifyContent: "space-around",
                    alignItems: "baseline",
                    flexDirection: "column",
                  }}
                >
                  <a style={{ margin: 20, fontSize: 25, color: "white" }}>
                    <b>{item.name}</b>
                  </a>
                  <Button
                    href="./Components/DetailPage.js"
                    target="_self"
                    variant="outlined"
                    style={{ color: "white" }}
                  >
                    View Detail
                  </Button>
                </div>
              </div>
            );
          })}
        <Pagination
          className="pagination-bar"
          currentPage={currentPage}
          totalCount={
            pokemons.pokeItems.results ? pokemons.pokeItems.results.length : 0
          }
          pageSize={PageSize}
          onPageChange={(page) => setCurrentPage(page)}
        />
      </div>
    </div>
  );
}
export default App;
