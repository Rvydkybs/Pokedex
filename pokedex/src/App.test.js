import App from "./App";
import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

//testing react native apps is the same way with this,but just the testing library name is different
//we can write test.js for any component we want

test("renders first test", () => {
  //declare this test structure at the first time,this string is title of out test function
  const comp = render(<App />); //the component which is gonna be testing
  /*expect: the object that we need to get answers when its tested ,we can write anything inside expect(),exmp:5+9,
  expect.function(); Does expect object is meet the condition ?*/
  expect(comp).toMatchSnapshot(); //is component match with snapshot
});

test("example by test id", () => {
  //declare this test structure at the first time,this string is title of out test function
  const comp = render(<App />);
  // const title = comp.getByTestId("title");
  // console.log("test id element:", title);
});
