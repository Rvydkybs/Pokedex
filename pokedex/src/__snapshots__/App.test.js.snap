// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="background-color: rgb(238, 238, 238); flex: 1; display: flex; flex-direction: column;"
      >
        <h1>
          <b>
            Pokedex
          </b>
        </h1>
        <div
          style="width: 100%; flex-direction: column; display: flex; align-items: center;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="background-color: rgb(238, 238, 238); flex: 1; display: flex; flex-direction: column;"
    >
      <h1>
        <b>
          Pokedex
        </b>
      </h1>
      <div
        style="width: 100%; flex-direction: column; display: flex; align-items: center;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders first test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="background-color: rgb(238, 238, 238); flex: 1; display: flex; flex-direction: column;"
      >
        <h1>
          <b>
            Pokedex
          </b>
        </h1>
        <div
          style="width: 100%; flex-direction: column; display: flex; align-items: center;"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="background-color: rgb(238, 238, 238); flex: 1; display: flex; flex-direction: column;"
    >
      <h1>
        <b>
          Pokedex
        </b>
      </h1>
      <div
        style="width: 100%; flex-direction: column; display: flex; align-items: center;"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
